pipeline-examples/trigger-job-on-all-nodes/triggerJobOnEveryNode.groovy

```diff
@@ -0,0 +1,27 @@
+// The script triggers PayloadJob on every node.
+// It uses Node and Label Parameter plugin to pass the job name to the payload job.
+// The code will require approval of several Jenkins classes in the Script Security mode
+def branches = [:]
+def names = nodeNames()
+for (int i=0; i<names.size(); ++i) {
+  def nodeName = names[i];
+  // Into each branch we put the pipeline code we want to execute
+  branches["node_" + nodeName] = {
+    node(nodeName) {
+      echo "Triggering on " + nodeName
+      build job: 'PayloadJob', parameters: [
```

Comment:
Mmm, not sure. IIUC you are grabbing a node (`node(nodeName)` just to call `build` so `PayloadJob` is built on `nodeName`). As I see it the `node(nodeName)` is not needed (even counterproductive).